WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -107,9 +150,46 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
     func didTapEditAd() {
         onEditAd?()
     }
+
+    func loadAISuggestions() async {
+        isLoadingAISuggestions = true
+        errorState = .none
+
+        do {
+            suggestions = try await fetchAISuggestions()
+            if let firstSuggestion = suggestions.first {
+                tagline = firstSuggestion.siteName
+                description = firstSuggestion.textSnippet
+            }
+        } catch {
+            DDLogError("⛔️ Error fetching Blaze AI suggestions: \(error)")
+            errorState = .fetchingAISuggestions
+        }
+
+        isLoadingAISuggestions = false
+    }
+
+    func downloadProductImage() async {
+        isDownloadingImage = true
+        if let productImage = await loadProductImage() {
+            image = productImage
+        }
+        isDownloadingImage = false
+    }
 }
 
 private extension BlazeCampaignCreationFormViewModel {
+    func loadProductImage() async -> MediaPickerImage? {
+        await withCheckedContinuation({ continuation in
+            guard let firstImage = product?.images.first else {
+                return continuation.resume(returning: nil)
+            }
+            _ = productImageLoader.requestImage(productImage: firstImage) { image in
```

Comment:
question: do you think it's worth it to log it if requestImage() somehow fails?

Reply:
Good point. Unfortunately, the `requestImage` doesn't return an error at all. I will skip changing that now as it gets out of scope.